### PR TITLE
Typespec on helpers

### PR DESCRIPTION
**Pull request by Tristramg**

### CODE REVIEW

File: apps/helpers/lib/hasher.ex [hunk 1]
```diff
@@ -38,11 +38,13 @@ defmodule Hasher do
     end
   end
 
+  @spec update_hash(binary() | :error, binary() | :error) :: binary() | :error
   defp update_hash(_, :error), do: :error
   defp update_hash(:error, _), do: :error
   defp update_hash(chunk, hash) when is_binary(chunk), do: :crypto.hash_update(hash, chunk)
```
Comment (antoine-de) [inline apps/helpers/lib/hasher.ex]: why is the `when is_binary(chunk)` needed?

File: apps/helpers/lib/http_stream.ex [hunk 2]
```diff
@@ -5,6 +5,7 @@ defmodule HTTPStream do
   require Logger
   alias Mint.HTTP
 
+  @spec get(any()) :: ({:cont, any()} | {:halt, any()} | {:suspend, any()}, any() -> any())
```
Comment (antoine-de) [inline apps/helpers/lib/http_stream.ex]: isn't the url a binary?

Comment (antoine-de) [reply apps/helpers/lib/http_stream.ex]: or `binary | URI.t()`?

File: apps/helpers/lib/hasher.ex (hunk 1 above)
Comment (Tristramg) [reply apps/helpers/lib/hasher.ex]: I don’t think it is needed. Neither should the first parameter ever be :error, but I’m a bit afraid to touch it. I don’t trust that much the typing system. What do you think?

File: apps/helpers/lib/http_stream.ex (hunk 2 above)
Comment (Tristramg) [reply apps/helpers/lib/http_stream.ex]: :+1:

File: apps/helpers/lib/hasher.ex (hunk 1 above)
Comment (Tristramg) [reply apps/helpers/lib/hasher.ex]: Meh. Let’s be optimistic about the stream library and dialyzer ;)

**Pull request merged**